asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     custom_cell_magics: kql
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.2
+#   kernelspec:
+#     display_name: installersurvey
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# ## Section 6: Maintenance and servicing
+# **Aims:** Generating outputs for Section 6 (RQs 6.01-6.25) as planned in [Heat Pump Installer Survey data analysis plan](https://docs.google.com/document/d/1M1nzdf3fyTjipmaKJKViin0EB3e3R1afOQglwMGJaII/edit#heading=h.jzmlm8j0kyve)<br>
+# **Author:** Elysia Lucas<br>
+# **Date:** 2024-04-04<br>
+# <i>Update:</i> 2024-04-17 (Re-coding of free text responses)
+# <p> 
+# N.B. As of 11.04.2024, only CompanySizeOwner and CompanySizeEmployee outputs with version 2 categorisation are generated.
+
+# %% [markdown]
+# ### Importing and pre-processing
+
+# %%
+with open("data_setup.py") as file:
+    exec(file.read())
+
+with open("free_text_recode.py") as file:
+    exec(file.read())
+
+# %%
+from matplotlib import pyplot as plt
+
+from IPython.core.interactiveshell import InteractiveShell
+InteractiveShell.ast_node_interactivity = "all"
+
+# %% [markdown]
+# ### RQ 6.01 What proportion of survey participants offer repairs, servicing and maintenance to heat pump customers
+# - SQ: 42a, 42b
+# - Sample: Standard_output (employment type categorisation)
+
+# %%
+# Print possible answers
+data[col.q42a].value_counts()
+data[col.q42b].value_counts()
+
+
+# %%
+def condition_q42(x):
```

Comment:
This is a bit picky but type hints really help me when reading code!
```suggestion
from typing import Dict
def condition_q42(x: Dict[str, str]):
```